### ISSUE TITLE
Graphs failing to visualize all the data.

### ISSUE THREAD

**Issue by neilh10** [ODM2/ODM2DataSharingPortal#489]
All the data from a Mayfly integration test POSTED to  
https://monitormywatershed.org/sites/tu_rc_test04/
is not showing on the graphs, both sparkline and TSA show the same problem.

I would expect that the visualizations/graphs would show the complete set of data, in time order, even if delivered out of order.

The POSTs where created in a setting where the battery voltage was dropping, and delivery of the readings was halted until more power (Solar) had been collected. 
Then when there was sufficient power accumulated, the latest 2 readings where POSTed, followed by up to 11 readings from a FIFO of previous readings.  After 11 readings, a check was done on the battery level, and if there was insufficient power, no more POSTS.  This effectively delayed the posts till the next attempt to connect to the internet.

The test system was set for taking a reading every 2 minutes, and POSTing once two readings are collected. POST attempts are made every four minutes. 
A sequence number "EnviroDIY_Mayfly_SampleNum" was maintained, from 1 to 1198 and shows no missing readings.  
The file tu_rc_test04_TimeSeriesResults_201024_1347.xlsx contains data downloaded from 
https://monitormywatershed.org/sites/tu_rc_test04/
 The data relevant to this series of POSTS is in the attached file
[tu_rc_test04_TimeSeriesResults_201024_1347.xlsx](https://github.com/ODM2/ODM2DataSharingPortal/files/5433871/tu_rc_test04_TimeSeriesResults_201024_1347.xlsx)

The EnviroDIY_Mayfly_SampleNum shows no missing samples, so all data was successfully POSTed and recorded. 
The sparkline graph below for some reason has two unexpected gaps in them, that don't exist in the downloaded data.
![201024_1511tu_rc_test04sparklinePlot](https://user-images.githubusercontent.com/2766619/97094847-c034be00-160d-11eb-937d-8d613c2c4029.jpg)

**Comment by neilh10** [comment]
a data point from this mornings testing.
For the test system 
https://monitormywatershed.org/sites/tu_rc_test06/
The Mayfly was paused in transmitting data as battery running low, (a normal condition). Once the solar returned and started transmitting, it transmitted the latest readings first, followed by earlier readings not uploaded, (out-of-order) . 
These all show  differently between the different data views. a) Front page Sparkline plot, b) sparkline number, c) TSA visualizations, and d) downloaded data.
Using the SampleNum, the downloaded data appears to be accurate.

**Comment by neilh10** [comment]
Thanks for the updates and timelines. Woohooh a target date!!
Would you like me to use https://staging.monitormywatershed.org  for some testing or is it too soon.?
I have just started an accelerated stability test (not live data)  using an WiFi S6, I could switch it to https://staging.monitormywatershed.org . Sampling is set to every 2 minutes, collect 4 readings before upload, so attempt upload every 8minutes, and attempt send 4samples each upload. With the WiFi its easy to simulate communications failure, turn off the wifi, allow readings to collect, and then when the WiFi is turned back on it attempts a bulk upload.   
I've also changed the WiFi upload algorithm, so it sets up a tcp/link and then attempts to bulk upload ALL the readings, before tearing it down. 
Before, (on WiFi and cellular the TinyGSM lib)  seemed like on a bulk upload attempt, the Mayfly algorithm was tearing down the TCP/IP link, after each POST and response, and then setting up a new link for the next POST .

**Comment by aufdenkampe** [comment]
We have decided to replace TSA with a Time Series Visualization (TSV) page, starting with:
- #512

Check out the screenshots I just posted with https://github.com/ODM2/ODM2DataSharingPortal/issues/512#issuecomment-957935317. 

I'm 100% sure new TSV addresses this issue, as we have removed the troublesome translation layers (i.e. TSA catalog and InfluxDB) that have caused so many of our problems, and the new TSV fetches data directly from the main ODM2 database.

We'll be posting all this to  https://staging.monitormywatershed.org by the end of next week. I'll keep this open until then to confirm.
